src/rshim.c

```diff
@@ -2092,22 +2141,23 @@ int rshim_set_drop_mode(rshim_backend_t *bd, int value)
     bd->drop_pkt = 1;
   else
     rshim_fifo_sync(bd, true);
-  pthread_mutex_unlock(&bd->mutex);
   /*
    * Check if another endpoint driver has already attached to the
    * same rshim device before enabling it.
    */
   if (!bd->drop_mode) {
     rshim_lock();
-    pthread_mutex_lock(&bd->mutex);
     if (rshim_access_check(bd)) {
       RSHIM_WARN("rshim%d is not accessible\n", bd->index);
       bd->drop_mode = 1;
       rt = -EACCES;
     }
-    pthread_mutex_unlock(&bd->mutex);
     rshim_unlock();
   }
+
+  if (has_bd_lock)
+    pthread_mutex_unlock(&bd->mutex);
+
```

Comment:
Can we change these lines back? It's done in such way for a reason to avoid potential dead lock.

Reply:
better not to change the locking order between rshim_lock() and pthread_mutex_unlock(), which could potentially cause deadlock.

Reply:
Changes pushed in e605c95